kdoctor/src/commonMain/kotlin/org/jetbrains/kotlin/doctor/diagnostics/GradleProjectDiagnostic.kt

```diff
@@ -0,0 +1,186 @@
+
+        if (gradle == null) {
+            result.addFailure(
+                "Gradle is not found",
+                "For better experience we recommend to use a Gradle wrapper"
+            )
+            return result.build()
+        } else {
+            result.addSuccess(
+                """${gradle.name} (${gradle.version})"""
+            )
+        }
+
+        val tempFile = try {
+            system.writeTempFile(printPluginsInitScript())
+        } catch (e: Exception) {
+            result.addFailure(
+                "Error: impossible to create temporary file",
+                "Check your file system write permissions",
+                "'${e.message}'"
+            )
+            return result.build()
+        }
+        val initScript = tempFile.substringBeforeLast("/") + "/init.gradle.kts"
+
+        system.execute("rm", initScript) //result code doesn't matter
+
+        val initScriptRenamed = system.execute("mv", tempFile, initScript).code
+        if (initScriptRenamed != 0) {
+            result.addFailure(
+                "Error $initScriptRenamed: impossible to prepare $initScript file",
+                "Check your file system write permissions"
+            )
+            return result.build()
+        }
+
+        Logger.d("Created $initScript")
+
+        val runGradle = system.execute(gradleBin, "-p", projectPath, "-I", initScript)
+
+        if (runGradle.code != 0) {
+            result.addFailure(
+                "Gradle error",
+                "Run in terminal '$gradleBin -p $projectPath --info'"
+            )
+            return result.build()
+        }
+
+        val gradleOutput = runGradle.output.orEmpty()
+        val pluginArtifacts = mutableListOf<PluginArtifact>()
+        val appliedPluginsMap = mutableMapOf<String, String>()
+        gradleOutput.lineSequence().forEach { l ->
+            if (l.contains(KDOCTOR_PREFIX)) {
+                val line = l.substringAfter(KDOCTOR_PREFIX)
+                if (line.startsWith("DEP ")) {
+                    //build script dependency
+                    val (group, name, version) = line.substringAfter("DEP ").split(":")
+                    pluginArtifacts.add(PluginArtifact(group, name, version))
+                } else {
+                    val (pluginId, pluginVersion) = line.split("=")
+                    //a plugin can be applied without a version
+                    if (!appliedPluginsMap.containsKey(pluginId) || pluginVersion != "null") {
+                        appliedPluginsMap[pluginId] = pluginVersion
+                    }
+                }
+            }
+        }
+
+        val appliedPlugins = findPluginVersions(appliedPluginsMap, pluginArtifacts).sortedBy { it.name }
+        if (appliedPlugins.isNotEmpty()) {
+            result.addSuccess(
+                "Gradle plugins:",
+                *appliedPlugins.map { it.name + ":" + it.version }.toTypedArray()
+            )
+        } else {
+            result.addWarning(
```

Comment:
Shouldn't the case when there is no multiplatform plugin be an error?

Reply:
It is not an error IMHO. just a non mpp project